### ISSUE TITLE
Include a hostname tag in metrics

### ISSUE THREAD

**Issue by pjenvey** [mozilla-services/contile#200]
So we can group some metrics by the k8s pod let's include the hostname in metrics as a tag. We probably only need the tag for certain metrics, most importantly the ones emitted by the cache's periodic reporter

**Comment by jrconlin** [comment]
@pjenvey As I understand, the desire is to determine the pod name for a given set of metrics. From [this article](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/) it appears that the pod name is not something that's generally available, and may not be something that can be easily determined from a given hostname. 

Operations (@jbuck) has expressed concerns about potential cardinality issues around including hostnames.

To that end:

1) What info are we looking to capture?
  `spec.nodeName`? `metadata.name`? `metadata.namespace`?, `???`?
2) How would that data be correlated and reported with the metrics we're capturing?
3) Do we need that data only for metric reporting, or do we want it for both both sentry and metrics?

**Comment by pjenvey** [comment]
At least in syncstorage-rs k8s' `pod_name` matches the `hostname`. See sync's prod dashboard CPU utilization panel (from k8s, GROUP BY tag(pod_name)) vs the Spanner Active Connections panel (syncstorage-rs's metrics, GROUP BY tag(hostname)) -- same values.

**Comment by jrconlin** [comment]
Ok, we should get @jbuck to confirm that's true and that syncstorage isn't doing something special. 

Is it also ok if this tag is only added to the cadence load or should we add it to potential sentry events too?

**Comment by pjenvey** [comment]
> Is it also ok if this tag is only added to the cadence load or should we add it to potential sentry events too?

It's not needed in sentry as their events already have it in their own `server_name` field automatically filled by the sentry client.

**Comment by jbuck** [comment]
> Ok, we should get @jbuck to confirm that's true and that syncstorage isn't doing something special.

Yeah, that makes sense, I don't think syncstorage is doing anything special

**Comment by data-sync-user** [comment]
➤ Ron Manning commented:

Jon Buckley Do you have time to review this?

**Comment by data-sync-user** [comment]
➤ JR Conlin commented:

PR: https://github.com/mozilla-services/contile/pull/234 ( https://github.com/mozilla-services/contile/pull/234|smart-link )
